File: problems/3216-lexicographically-smallest-string-after-a-swap/solution.cpp
class Solution {
public:
    string getSmallestString(string s) {
        int n = s.length();
        for (int i = 0; i < n; ++i) {
            for (int j = i + 1; j < n; ++j) {
                if ((s[i] - '0') % 2 == (s[j] - '0') % 2 && s[j] < s[i]) {
                    swap(s[i], s[j]);
                    break;
                }
            }
        }
        return s;
    }
};

File: problems/3216-lexicographically-smallest-string-after-a-swap/solution.java
class Solution {
    public String getSmallestString(String s) {
        char[] chars = s.toCharArray();
        int n = chars.length;

        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if ((chars[i] - '0') % 2 == (chars[j] - '0') % 2 && chars[j] < chars[i]) {
                    char temp = chars[i];
                    chars[i] = chars[j];
                    chars[j] = temp;
                }
            }
        }

        return new String(chars);
    }
}